### PR TITLE
Add GUC for ORCA Eager Agg and bump to 3.27.0

### PR DESCRIPTION
**Pull request by iyerr3**
Eager Agg is an experimental feature in ORCA and is gated using a new GUC: `optimizer_enable_eageragg` with default value as `off`.

### CODE REVIEW

**Comment by hlinnaka** [comment]
What's the status of this PR? Do we still want this, or should this just be closed?

**Comment by berlin-ab** [comment]
I spoke to @iyerr3. He's not currently focusing on this work at the moment. Is this still something we want?

**Comment by bhuvnesh2703** [comment]
@hlinnaka @berlin-ab We plan to push it in this week or the next week..

File: src/backend/gpopt/config/CConfigParamMapping.cpp [hunk 1]
```diff
@@ -377,6 +377,13 @@ CConfigParamMapping::SConfigMappingElem CConfigParamMapping::m_elements[] =
 		&optimizer_force_agg_skew_avoidance,
 		false, // m_negate_param
 		GPOS_WSZ_LIT("Always pick a plan for aggregate distinct that minimizes skew.")
+        },
+
+        {
+		EopttraceEnableEagerAgg,
+		&optimizer_enable_eageragg,
+		false, // m_negate_param
+		GPOS_WSZ_LIT("Enable Eager Agg Xform for pushing aggregate below an innerjoin.")
```
Comment (berlin-ab) [inline src/backend/gpopt/config/CConfigParamMapping.cpp]: I don't know anything about this area of the codebase, but what is an XForm? Is this text user facing? If so, is there something more user friendly than Eager Agg Xform?

File: src/backend/utils/misc/guc_gp.c [hunk 2]
```diff
@@ -3039,6 +3040,17 @@ struct config_bool ConfigureNamesBool_gp[] =
 		},
 		&gp_enable_global_deadlock_detector,
 		false, NULL, NULL
+    },
+
+	{
+		{"optimizer_enable_eageragg", PGC_USERSET, DEVELOPER_OPTIONS,
+			gettext_noop("Enable Eager Agg Xform for pushing aggregate below an innerjoin."),
```
Comment (berlin-ab) [inline src/backend/utils/misc/guc_gp.c]: same comment as above.

File: src/backend/gpopt/config/CConfigParamMapping.cpp (hunk 1 above)
Comment (bhuvnesh2703) [reply src/backend/gpopt/config/CConfigParamMapping.cpp]: Xform => you can equate it to a code block performing a transformation. In ORCA, all the transformation rules in exploration / implementation phase are implemented using a Transform function (i.e short Xform)
This is a developer GUC and not exposed for users (i.e not documented)

Comment (berlin-ab) [reply src/backend/gpopt/config/CConfigParamMapping.cpp]: Thanks.

Comment (bhuvnesh2703) [reply src/backend/gpopt/config/CConfigParamMapping.cpp]: and made the change from Xform to transform.

**Pull request merged**